Anvil/Tools/Actions.pm

```diff
@@ -0,0 +1,289 @@
+# sense in this case to think of it as a parent.
+sub parent
+{
+	my $self   = shift;
+	my $parent = shift;
+	
+	$self->{HANDLE}{TOOLS} = $parent if $parent;
+	
+	# Defend against memory leads. See Scalar::Util'.
+	if (not isweak($self->{HANDLE}{TOOLS}))
+	{
+		weaken($self->{HANDLE}{TOOLS});
+	}
+	
+	return ($self->{HANDLE}{TOOLS});
+}
+
+
+#############################################################################################################
+# Public methods                                                                                            #
+#############################################################################################################
+
+=head2 insert_action_node_assume
+
+This method inserts "1" into "scancore::actions::node::{number}::assume" variable's value where number is parsed from node_name. 
+This method also updates "scancore::actions::node::{number}::assume" variable's value to "0" where number is parsed from peer_node_name. This insures that there is only one assume action, two assumes do not make sense.
+
+If there is an error, C<< !!error!! >> is returned.
+
+Parameters;
+
+=head3 node_name (required)
+
+This is the name/number of the node on which assume action was executed by ScanCore. Must include the node number that is either "1" or "2". Example: C<< node1 >> or C<< 2 >>
+
+=head3 peer_node_name (required)
+
+This is the name/number of the peer node to unset its assume variable. Must include the node number that is either "1" or "2". Example: C<< node1 >> or C<< 2 >>
+
+=cut
+sub insert_action_node_assume {
+	my $self      = shift;
+	my $parameter = shift;
+	my $anvil     = $self->parent;
+	my $debug     = defined $parameter->{debug} ? $parameter->{debug} : 2;
+	$anvil->Log->entry({source => $THIS_FILE, line => __LINE__, level => $debug, key => "log_0125", variables => { method => "Actions->insert_action_node_assume()" }});
+
+	my $node_name      = defined $parameter->{node_name} ? $parameter->{node_name} : "";
+	my $peer_node_name = defined $parameter->{peer_node_name} ? $parameter->{peer_node_name} : "";
+	$anvil->Log->variables({source => $THIS_FILE, line => __LINE__, level => $debug, list => { 
+		node_name      => $node_name,
+		peer_node_name => $peer_node_name,
+	}});
+
+	if (not $node_name)
+	{
+		$anvil->Log->entry({source => $THIS_FILE, line => __LINE__, level => 0, priority => "err", key => "log_0020", variables => { method => "Actions->insert_action_node_assume()", parameter => "node_name" }});
+		return("!!error!!");
+	}
+
+	if (not $peer_node_name)
+	{
+		$anvil->Log->entry({source => $THIS_FILE, line => __LINE__, level => 0, priority => "err", key => "log_0020", variables => { method => "Actions->insert_action_node_assume()", parameter => "peer_node_name" }});
+		return("!!error!!");
+	}
+
+	# Can we parse a node number from node_name or peer_node_name
+	my ($node_number) = $node_name =~ /(\d)/;
```

Comment:
We don't require nodes to have specific names. We figure out which node is node 1 and 2 by comparing the host's host_uuid to the "anvil_node{1,2}_host_uuid" in the anvils table. If you call $anvil->Database->get_anvils() and then check if the host's host_uuid is in '$anvil->data->{anvils}{anvil_uuid}{$anvil_uuid}{anvil_node1_host_uuid}' or '$anvil->data->{anvils}{anvil_uuid}{$anvil_uuid}{anvil_node2_host_uuid}'.